packages/components/addon/components/hds/side-nav/list/title.hbs

```diff
@@ -0,0 +1,5 @@
+<Hds::SideNav::List::Item>
+  <div class="hds-side-nav__list-title hds-typography-body-100 hds-font-weight-semibold">
+    {{yield}}
+  </div>
```

Comment:
@KristinLBradley something I noticed in the Cloud UI version of the `SideNav`: they strip away the whitespace for the "header/title", probably better if we do the same (they must have had good reasons)

<img width="293" alt="screenshot_2462" src="https://user-images.githubusercontent.com/686239/223856680-7cf06be0-fda1-4758-9ab7-58cb99455729.png">

```suggestion
  <div class="hds-side-nav__list-title hds-typography-body-100 hds-font-weight-semibold">{{~yield~}}</div>
```